users/tests/test_unit_validators.py

```diff
@@ -44,3 +48,41 @@ def _assert_date_of_birth_is_valid(self, date_of_birth):
             self.user_age_validator(date_of_birth)
         except Exception as e:  # pylint:disable=broad-except
             self.fail(f"Unexpected exception {str(e)} has occurred!")
+
+
+class TestUserEmailValidation(TestCase):
+    def setUp(self) -> None:
+        self.user_email_validation = UserEmailValidation()
+
+    @parameterized.expand(
+        [
+            ("testusercodepoets.it", ValidationErrorText.VALIDATION_ERROR_EMAIL_AT_SIGN_MESSAGE),
+            ("@codepoets.it", ValidationErrorText.VALIDATION_ERROR_EMAIL_MALFORMED_FIRST_PART),
+            ("testuser@invaliddomain.com", ValidationErrorText.VALIDATION_ERROR_EMAIL_MESSAGE_DOMAIN),
+        ]
+    )
+    def test_validator_should_throw_exception_with_invalid_email(self, email, exception_message):
+        with self.assertRaises(ValidationError) as exception:
+            self.user_email_validation(email)
+        self.assertEqual(exception_message, exception.exception.message)
+
+    @parameterized.expand(
+        [
+            ("valid@codepoets.it", ["codepoets.it"]),
+            ("valid@otherdomain.it", ["otherdomain.it"]),
+            ("valid@fewdomain.it", ["test.pl", "fewdomain.it"]),
+        ]
+    )
+    def test_validator_should_allow_all_email_domains_with_empty_valid_domain_list(self, email, valid_email_domains):
+        try:
+            with override_settings(VALID_EMAIL_DOMAIN_LIST=valid_email_domains):
+                valid_email = self.user_email_validation(email)
+        except ValidationError as exception:
+            self.fail(exception.messages)
+        self.assertEqual(valid_email, email)
```

Comment:
You don't have to add here `try...except` block. If method raise exception test will fail anyway.
You can make it shorter:

```python
        with override_settings(VALID_EMAIL_DOMAIN_LIST=valid_email_domains):
            valid_email = self.user_email_validation(email)
        self.assertEqual(valid_email, email)
```

Reply:
Done